### ISSUE TITLE
NewAuth usage

### ISSUE THREAD

**Issue by Fodoj** [sendgridlabs/go-kinesis#23]
I can't figure out how to new `NewAuth()` function together with `NewClient()`. Could you provide an example please? This, of course, doesn't work"

```go
kinesis_client = kinesis.New(kinesis.NewAuth(), kinesis.Region{os.Getenv("KINESIS_REGION")})
```

It gives me `panic: runtime error: invalid memory address or nil pointer dereference`, though compiles okay

**Comment by le0pard** [comment]
`ksis := kinesis.New(&kinesis.Auth{}, kinesis.Region{})`

**Comment by Fodoj** [comment]
@le0pard It doesn't invoke `NewAuth` and thus env variables are never used

**Comment by le0pard** [comment]
So, kinesis cli doesn't work, does it?

**Comment by Fodoj** [comment]
@le0pard I didn't test it. But I can't understand how it can work, if it `NewAuth` accepts zero arguments and then tries to do something with `auth` variable. I can provide pull request with a fix, if it's really a bug.

**Comment by le0pard** [comment]
@Fodoj will be good, because right now I couldn't test it.

**Comment by le0pard** [comment]
closed by #24
